phoenix-core/src/main/java/org/apache/phoenix/compile/StatementContext.java

```diff
@@ -98,11 +98,18 @@ public StatementContext(PhoenixStatement statement, ColumnResolver resolver, Sca
         this.aggregates = new AggregationManager();
         this.expressions = new ExpressionManager();
         PhoenixConnection connection = statement.getConnection();
-        this.dateFormat = connection.getQueryServices().getProps().get(QueryServices.DATE_FORMAT_ATTRIB, DateUtil.DEFAULT_DATE_FORMAT);
-        this.dateFormatTimeZone = TimeZone.getTimeZone(
-                connection.getQueryServices().getProps().get(QueryServices.DATE_FORMAT_TIMEZONE_ATTRIB, DateUtil.DEFAULT_TIME_ZONE_ID));
+        if (connection!=null) {
```

Comment:
Not crazy about having to do this null check as it may cascade to a lot of places. You may be able to create a PhoenixConnection from IndexMaintainer if need be.